src/linkedList.js

```diff
@@ -1,152 +1,57 @@
 class Node {
-  constructor( data, next = null ) {
-    this.data = data
-    this.next = next
+  constructor(data) {
+    this.data = data,
+    this.next = null
   }
-  data() { return this.data }
-
-  next() { return this.next }
 }
 
-export default class LinkedList {
-  constructor() {
-    this.head = null
-    this._length = 0
+export default class linkedList {
+  constructor(){
+    this._length = 0,
+    this.head = null,
+    this.tail = null
   }
 
-  insert( data ) {
-    let node = new Node( data )
-    let currentNode = this.head
+  insert(data) {
+    let node = new Node(data)
 
-    if( !this.head ) {
-      this.head = node
+    if(this.head === null){
       this._length++
-      return node
-    }
-    while( currentNode.next ) {
-      currentNode = currentNode.next
+      return this.head = this.tail = node
     }
-    currentNode.next = node
-    // console.log('data',node);
-    this._length++
-    return node
-
-  }
-
-  getHeadNode() { return this.head }
-
-  getTailNode() {
     let currentNode = this.head
-    console.log(currentNode)
-    // currentNode.next = this.next
 
-    while( currentNode.next ) {
+    while(currentNode.next){
       currentNode = currentNode.next
     }
-
-    return currentNode
-  }
-
-  contains( data ) {
-    let currentNode = this.head
-
-    while( currentNode.next ) {
-      if( currentNode.data === data ){
-        return true
-      } else {
-        currentNode = currentNode.next
-      }
-    }
-
-    return false
-  }
-
-  find( data ) {
-    if(this.head){
-      let currentNode = this.head
-
-      while( currentNode.next ) {
-        // console.log(currentNode)
-        if( currentNode.data === data ){
-          return currentNode
-        } else {
-          currentNode = currentNode.next
-        }
-      }
-      return -1
-    }else{
-      return -1
-    }
-  }
-
-  insertFirst( data ) {
-    let node = new Node( data )
-    let currentNode = this.head
-
-    node.next = currentNode
-    this.head = node
-
-    return node
+    this._length++
```

Comment:
++this._length

from stackoverflow: http://stackoverflow.com/questions/3469885/somevariable-vs-somevariable-in-javascript
++x (pre-increment) means "increment the variable; the value of the expression is the final value"
x++ (post-increment) means "remember the original value, then increment the variable; the value of the expression is the original value"

are you choosing this._length++ deliberately?